src/ResourceManagement/ResourceManager/Domain/IResourceGroups.cs

```diff
@@ -26,14 +27,29 @@ public interface IResourceGroups  :
         /// </summary>
         /// <param name="name">name The name of the resource group to check. The name is case insensitive</param>
         /// <returns>true if the resource group exists; false otherwise</returns>
+        [Obsolete("Use Contain() instead.")]
         bool CheckExistence (string name);
 
         /// <summary>
         /// Checks whether resource group exists.
         /// </summary>
         /// <param name="name">name The name of the resource group to check. The name is case insensitive</param>
         /// <returns>true if the resource group exists; false otherwise</returns>
-        Task<bool> CheckExistenceAsync(string name, CancellationToken cancellationToken = default(CancellationToken)
-);
+        bool Contain(string name);
+
+        /// <summary>
+        /// Checks whether resource group exists.
+        /// </summary>
+        /// <param name="name">name The name of the resource group to check. The name is case insensitive</param>
+        /// <returns>true if the resource group exists; false otherwise</returns>
+        [Obsolete("User ContgainAsync() instead.")]
+        Task<bool> CheckExistenceAsync(string name, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Checks whether resource group exists.
+        /// </summary>
+        /// <param name="name">name The name of the resource group to check. The name is case insensitive</param>
+        /// <returns>true if the resource group exists; false otherwise</returns>
+        Task<bool> ContainAsync(string name, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
>ContainAsync [](start = 19, length = 12)

LINQ analogy suggests plural - Contains. [https://msdn.microsoft.com/en-us/library/bhkz42b3(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/bhkz42b3(v=vs.110).aspx)

Reply:
i.e. instead of saying "Resource groups contain" I say "Resource group list contains". just saying :) 

---
In reply to: [145505367](https://github.com/Azure/azure-libraries-for-net/pull/66#discussion_r145505367) [](ancestors = 145505367)

Reply:
I've been debating this myself, but just looking at the context where this will be used:
`resourceGroups().contains()` just sounds awkward (not grammatically correct).
It's another thing when we talk about a String (`myString.Contains("foo")`) which sounds perfect. LINQ had to have general purpose naming, hanging off `IEnumerable`, so I can see why the singular form was preferable there.  
But doesn't it just feel icky where the only place one will be using this method is an object whose name is clearly plural (either as the accessor `.resourceGroups()` or the type `ResourceGroups`) and we'd still be using the singular form?